worker/worker.go

```diff
@@ -158,9 +158,9 @@ func (w *Worker) Transmit(addr string, id string, responseTo string, metadata ma
 	return
 }
 
-// EmitEvent emits a WorkerEvent, including an optional message.
-func (w *Worker) EmitEvent(event ipc.WorkerEventName, message string) error {
-	args := []interface{}{event}
+// EmitEvent emits a WorkerEvent, worker message id, and an optional message.
+func (w *Worker) EmitEvent(event ipc.WorkerEventName, message string, message_id string) error {
```

Comment:
Could we keep the order of arguments the same as the arguments of the D-Bus method? That reduces cognitive friction when translating between the D-Bus interface description and the implementations.